### PR TITLE
docs: dedupe gradient checkpointing page

### LINKED ISSUE #33432: Mostly duplicate checkpoint docs.
The pages https://docs.jax.dev/en/latest/notebooks/autodiff_remat.html and https://docs.jax.dev/en/latest/gradient-checkpointing.html seem extremely similar (I think many parts, such as the practical notes, are basically the same). It seems like these should be consolidated into one piece of documentation, unless I am missing some key difference between them?

### PR DESCRIPTION
**Pull request by tommasocerruti**
This PR fixes #33432.

The docs https://docs.jax.dev/en/latest/notebooks/autodiff_remat.html and https://docs.jax.dev/en/latest/gradient-checkpointing.html were extremely similar, so I made the notebook the canonical source, removed the duplicate page, added a redirect, and updated the toctree/anchors to keep links working.

Previous attempts:
* https://github.com/jax-ml/jax/pull/33497.  
* https://github.com/jax-ml/jax/pull/33499.  
were closed by @jakevdp because of not enough "bandwidth to review substantial documentation rewrites generated by an AI agent".

For this reason, I solved the issue with a redirect to keep the diff of the PR small and quick to review.

Testing: python3 -m sphinx -b html docs build/html

### CODE REVIEW

File: docs/gradient-checkpointing.md [hunk 1]
```diff
@@ -1,533 +0,0 @@
...
-f = chain_compose([jnp.sin] * 8)
-print_fwd_bwd(f, 3.)
-```
-
-```{code-cell}
-f = recursive_checkpoint([jnp.sin] * 8)
-print_fwd_bwd(f, 3.)
-```
-
-(gradient-checkpointing-practical-notes)=
```
[the remark this answers is not in the document]
Comment (tommasocerruti) [reply docs/gradient-checkpointing.md:482]: Good catch, I missed that. I now added the anchor to both the md and the ipynb.
I tested the change again by serving the docs locally, and the anchors worked well and redirected correctly.
